feat(portal): PHX-239: Patient portal project setup

flynncs opened the pull request:
### Changes

- New Patient Portal web package
- Deployment steps

### Deploys

- [x] **Deploy to Tamanu Internal** <!-- #deploy --> %opsref=patient-portal-pulumi

### Tests

- [ ] **Run E2E Tests** <!-- #e2e -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

flynncs committed:
loading up patient portal view

flynncs committed:
initial package setup w/ vite

flynncs committed:
fix build & move theming to shared

flynncs committed:
basic routing skeleton

flynncs committed:
import tamanu api

flynncs committed:
auth context & private / public routes

flynncs committed:
fix build

flynncs committed:
dockerfile / build update

flynncs committed:
include patient-portal in cd

flynncs committed:
Add CD workflow and config for patient portal package

Introduces a new GitHub Actions workflow for continuous deployment of the patient portal package, including artifact packaging and S3/release uploads. Updates the main CD workflow comment to include patient-portal, and extends ghaCdHelpers to support patient portal replica configuration.

flynncs committed:
Remove patient portal code from web package

Eliminated patient portal-specific views and routing from the web package, including the PatientPortalViews component and related environment variables and scripts. Updated patient-portal package to use a new environment variable (TAMANU_VITE_TARGET) and improved API endpoint configuration and proxy setup.

flynncs committed:
Refactor API context and usage in patient portal

Introduced ApiContext and useApi hook to provide the TamanuApi instance via React context. Updated Root and AuthProvider to use the new context, improving dependency management and testability. Adjusted TamanuApi to set the endpoint path to '/api'. Also updated Vite config to use process.env for proxy target and improved ESLint config for config files.

flynncs commented on .github/workflows/cd-package-patient-portal.yml:
would it be worth combining this with the `cd-package-frontend` yaml?

passcod replied:
Given they're otherwise identical, yes.

flynncs committed:
Consolidate frontend and patient-portal packaging workflows

Merged the patient-portal packaging workflow into the frontend workflow using a matrix strategy, and removed the separate cd-package-patient-portal.yml file. Refactored the Dockerfile to use shared build and runtime stages for both frontend and patient-portal, reducing duplication and improving maintainability.

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

flynncs committed:
swap to tsx

flynncs committed:
fix cd

tcodling commented on packages/patient-portal/public/fonts/Moul-Regular.ttf:
I am guessing we couldn't move these fonts and icons to shared?

flynncs replied:
just had a further look into this - seems like the only way to share the files in the `public` dir between the two web apps is to have a script copy the files on build or on dev, as we can't import them via JS
happy to do that if you think it's worth it?

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

passcod reviewed (approved):
✅ for .github changes only

flynncs committed:
add env.js to caddy

flynncs committed:
Add SPA fallback to Caddyfile for patient portal

Introduces a try_files directive to serve index.html for unmatched routes, enabling proper client-side routing support for the single-page application.

flynncs committed:
Remove ESLint config and dependencies from patient-portal

Deleted the custom eslint.config.js and removed all related ESLint dependencies from package.json and package-lock.json in the patient-portal package. Also updated React imports for consistency in several source files.

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

flynncs committed:
Centralize shared fonts and icons assets

Moved font and icon assets from patient-portal and web packages into a new shared/public directory. Updated references in HTML, manifest, and CSS files to use the new shared paths. Added symlinks in public directories for backward compatibility. This change reduces duplication and simplifies asset management across packages.

tcodling commented on packages/patient-portal/src/fonts.css:
```diff
@@ -0,0 +1,26 @@
+  src: url(shared/fonts/FiraCode-Medium.ttf) format('truetype');
+}
+
+@font-face {
+  font-family: 'Battambang';
+  font-style: normal;
+  font-display: swap;
+  src: url(shared/fonts/Battambang-Regular.ttf) format('truetype');
+  unicode-range: U+1780-17FF; /* Khmer characters */
+}
+
+@font-face {
+  font-family: 'Moul';
+  font-style: normal;
+  font-display: swap;
+  src: url(shared/fonts/Moul-Regular.ttf) format('truetype');
+  unicode-range: U+1780-17FF; /* Khmer characters */
+}
```
I think these have been removed from main so can probably be removed from here too

tcodling commented on packages/patient-portal/src/views/LoginView.tsx:
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Box, Typography, Container, Paper } from '@mui/material';
+
+export const LoginView = () => {
+  return (
+    <Container maxWidth="md">
+      <Box sx={{ mt: 4, mb: 4 }}>
+        <Paper sx={{ p: 3 }}>
+          <Typography variant="h4" component="h1" gutterBottom>
+            Patient Login
+          </Typography>
+          <Typography variant="body1" gutterBottom>
+            Login Page Content
+          </Typography>
+        </Paper>
+      </Box>
+    </Container>
+  );
```
Would be cool to maybe follow the styling patterns we use in the web package but totally fine to leave that to a later time as this is setup

tcodling reviewed (changes requested):
Looks amazing! Only thing is that you are referencing the old hardcoded khmer fonts that have been removed now. Its probably better to fix that up sooner rather than later. If you would rather fix inside the epic thats probably fine though just let me know

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

flynncs committed:
remove khmer font front theme

flynncs committed:
Remove redundant props and clean up PatientPrinting modals

Eliminated duplicate and unused getSetting props from various PatientPrinting modal components and improved destructuring for usePatientAdditionalDataQuery and useSurveyResponseQuery. Also simplified the useSelector callback in Settings context and removed redundant settings prop from SettingsProvider.

flynncs committed:
Revert "Remove redundant props and clean up PatientPrinting modals"

This reverts commit 4487d44dbb7decd7132beed9d32008c8f26c6bfa.

flynncs committed:
Merge branch 'epic-patient-portal' into feature/patient-portal-setup

tcodling reviewed (approved):
Thanks!

flynncs committed:
remove redundant fonts for portal
